backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/invalidation/impl/ShipmentScheduleInvalidateRepository.java

```diff
@@ -608,34 +626,56 @@ public void releaseRecomputeMarkerOutOfTrx(final PInstanceId pinstanceId)
 		final List<Object> sqlParams = ImmutableList.of(pinstanceId);
 		return TypedSqlQueryFilter.of(sql, sqlParams);
 	}
-	
+
 	@Override
 	public final void invalidateShipmentSchedulesFor(@NonNull final IQuery<I_M_ShipmentSchedule> shipmentScheduleQuery)
 	{
+		final String chunkUUID = UUID.randomUUID().toString();
 		final int count = shipmentScheduleQuery.insertDirectlyInto(I_M_ShipmentSchedule_Recompute.class)
-				.mapColumn(I_M_ShipmentSchedule_Recompute.COLUMNNAME_M_ShipmentSchedule_ID, 
-						I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID)
-				
+				.mapColumn(I_M_ShipmentSchedule_Recompute.COLUMNNAME_M_ShipmentSchedule_ID, I_M_ShipmentSchedule.COLUMNNAME_M_ShipmentSchedule_ID)
+				.mapColumn(I_M_ShipmentSchedule_Recompute.COLUMNNAME_C_Async_Batch_ID, I_M_ShipmentSchedule.COLUMNNAME_C_Async_Batch_ID)
+				.mapColumnToConstant(I_M_ShipmentSchedule_Recompute.COLUMNNAME_ChunkUUID, chunkUUID)
+
 				.execute()
 				.getRowsInserted();
 
 		logger.debug("Invalidated {} shipment schedules for {}", new Object[] { count, shipmentScheduleQuery });
 
-		
 		if (count > 0)
 		{
 			invalidateCacheForAllShipmentSchedules();
-			UpdateInvalidShipmentSchedulesWorkpackageProcessor.schedule();
+			enqueueShipmentScheduleRecompute(Env.getCtx(), ITrx.TRXNAME_ThreadInherited, chunkUUID);
 		}
 	}
-	
+
 	private void invalidateCacheForAllShipmentSchedules()
 	{
 		final IModelCacheInvalidationService modelCacheInvalidationService = Services.get(IModelCacheInvalidationService.class);
 
 		final CacheInvalidateMultiRequest multiRequest = CacheInvalidateMultiRequest.allRecordsForTable(I_M_ShipmentSchedule.Table_Name);
-		modelCacheInvalidationService.invalidate(multiRequest,  ModelCacheInvalidationTiming.CHANGE);
+		modelCacheInvalidationService.invalidate(multiRequest, ModelCacheInvalidationTiming.CHANGE);
 	}
-	
 
+	private void enqueueShipmentScheduleRecompute(
+			@NonNull final Properties context,
+			@Nullable final String trxName,
+			@NonNull final String chunkUUID)
```

Comment:
to me it's not clear what the chunkUUID is about. cosmin wrote
> we use it to make sure we are taking into consideration the same async batch ids that were pushed when the insert query ran

which sounds plausible but i still don't really get it. please make the code clearer and/or add comments and javadocs

Reply:
javadocs added